DNN Platform/DotNetNuke.ContentSecurityPolicy/DocumentCspContributor.cs

```diff
@@ -0,0 +1,114 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information
+
+namespace DotNetNuke.ContentSecurityPolicy
+{
+    using System;
+    using System.Linq;
+
+    /// <summary>
+    /// Contributor for document-level directives.
+    /// </summary>
+    public class DocumentCspContributor : BaseCspContributor
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DocumentCspContributor"/> class.
+        /// </summary>
+        /// <param name="directiveType">The directive type to create the contributor for.</param>
+        /// <param name="value">The value of the directive.</param>
+        public DocumentCspContributor(CspDirectiveType directiveType, string value)
+        {
+            this.DirectiveType = directiveType;
+            this.SetDirectiveValue(value);
+        }
+
+        /// <summary>
+        /// Gets value of the document directive.
+        /// </summary>
+        public string DirectiveValue { get; private set; }
+
+        /// <summary>
+        /// Sets the directive value with validation.
+        /// </summary>
+        /// <param name="value">The value to set for the directive.</param>
+        public void SetDirectiveValue(string value)
+        {
+            this.ValidateDirectiveValue(this.DirectiveType, value);
+            this.DirectiveValue = value;
+        }
+
+        /// <summary>
+        /// Generates the directive string.
+        /// </summary>
+        /// <returns>The directive string.</returns>
+        public override string GenerateDirective()
+        {
+            if (this.DirectiveType == CspDirectiveType.UpgradeInsecureRequests)
+            {
+                return $"{CspDirectiveNameMapper.GetDirectiveName(this.DirectiveType)}";
+            }
+
+            if (string.IsNullOrWhiteSpace(this.DirectiveValue))
+            {
+                return string.Empty;
+            }
+
+            return $"{CspDirectiveNameMapper.GetDirectiveName(this.DirectiveType)} {this.DirectiveValue}";
+        }
+
+        /// <summary>
+        /// Validates directive value based on directive type.
+        /// </summary>
+        private void ValidateDirectiveValue(CspDirectiveType type, string value)
+        {
+            switch (type)
+            {
+                case CspDirectiveType.PluginTypes:
+                    this.ValidatePluginTypes(value);
+                    break;
+                case CspDirectiveType.SandboxDirective:
+                    this.ValidateSandboxDirective(value);
+                    break;
+
+                    // Add more specific validations as needed
+            }
+        }
+
+        /// <summary>
+        /// Validates plugin types.
+        /// </summary>
+        private void ValidatePluginTypes(string value)
+        {
+            string[] validPluginTypes = { "application/pdf", "image/svg+xml" };
```

Comment:
Is this correct to hardcode only those mime-types, I am not sure but should this allow other types like word/excel, etc.?